packages/iov-core/src/servercore.ts

```diff
@@ -0,0 +1,80 @@
+import { ChainId, PublicIdentity, TransactionId, UnsignedTransaction } from "@iov/bcp-types";
+import { UserProfile, WalletId } from "@iov/keycontrol";
+
+import { MultiChainSigner } from "./multichainsigner";
+
+function publicIdentityEquals(left: PublicIdentity, right: PublicIdentity): boolean {
+  return (
+    left.chainId === right.chainId &&
+    left.pubkey.algo === right.pubkey.algo &&
+    left.pubkey.data.length === right.pubkey.data.length &&
+    left.pubkey.data.every((value, index) => value === right.pubkey.data[index])
```

Comment:
Nice compare. I think this should be more general in bcp-types or such. To compare Uint8Arrays and then compare PublicIdentities.

Reply:
Good point, implemented: https://github.com/iov-one/iov-core/pull/648/commits/c4187304ae6a4be01e7b3808ce79f2088e39e0eb